tests/sles4sap/ensa/ensa_filesystems.pm

```diff
@@ -18,60 +18,46 @@ use lockapi;
     my $nw_install_data = $self->netweaver_installation_data();
-    my $create_directories = join(',', 'sapmnt', 'SYS');
     my $sap_sid = $nw_install_data->{instance_sid};
     my $sap_dir = $nw_install_data->{sap_directory};
 
     select_serial_terminal;
 
-    # Setup NFS directory structure on supportserver side
-    if (check_var('SUPPORT_SERVER', '1')) {
-        my $nfs_permissions = get_required_var("NFS_PERMISSIONS");
-        record_info('NFS prep', 'Preparing SAP related exports');
-
-        assert_script_run("mkdir -p $nfs_root/$sap_sid/{$create_directories}");
-        # assert_script_run("chmod -Rv 777 $nfs_root/$sap_sid");
-        # assert_script_run("chown -Rv $sidadm_uid:$sapsys_guid $nfs_root/$sap_sid/*");
-        assert_script_run("echo $nfs_root *\($nfs_permissions\) >> /etc/exports");
-        assert_script_run("exportfs -r");
-        systemctl("restart nfs-server");
-        systemctl("restart rpcbind");
-        systemctl("is-active nfs-server -a rpcbind");
-        return;
-    }
-
-    # Mount shared SAP  NFS filesystems
-    record_info('NFS mounts', 'Preparing shared NFS filesystems');
-    assert_script_run("mkdir -p /sapmnt $sap_dir/{$create_directories}");
-    assert_script_run("echo 'ns:$nfs_root/$sap_sid/sapmnt /sapmnt nfs defaults 0 0' >> /etc/fstab");
-    assert_script_run("echo 'ns:$nfs_root/$sap_sid/SYS $sap_dir/SYS nfs defaults 0 0' >> /etc/fstab");
-    assert_script_run('mount -a');
-    assert_script_run("chmod -Rv 777 /sapmnt/");
-    assert_script_run("chmod -Rv 777 $sap_dir/");
-
     # Prepare and mount ISCSI devices
     my $instance_type = get_required_var('INSTANCE_TYPE');
     my $instance_dir = $nw_install_data->{instances}{$instance_type}{instance_dir_name};
     set_var('INSTANCE_ID', $nw_install_data->{instances}{$instance_type}{instance_id});
 
     if ($instance_type eq 'ASCS') {
-        record_info('ISCSI mounts', 'Preparing shared NFS filesystems');
         my $lun_path = get_lun(use_once => 0);    # ERS will use the same LUN.
         set_var('USR_SAP_DEVICE_ASCS', "$lun_path-part1");    # save it in variable for other modules
         set_var('USR_SAP_DEVICE_ERS', "$lun_path-part2");    # save it in variable for other modules
+
+        # Create the partitions
         assert_script_run("mkdir -p $sap_dir/$instance_dir");
+        assert_script_run(" parted -s $lun_path --list");
         assert_script_run("parted -s $lun_path mklabel gpt");
         assert_script_run("parted -s $lun_path mkpart primary 0% 50%");
+        script_run("partprobe -s");
         assert_script_run("parted -s $lun_path mkpart primary 50% 100%");
+        # From the manual: changes will *probably* be made to the disk
+        # immediately after typing a command. However, the operating system’s
+        # cache and the disk’s hardware cache may delay this. When using serial
+        # we hit this limitation, to avoid that we run partprobe and parted again.
+        script_run("partprobe -s");
```

Comment:
I think those should be single quotes as well.

Reply:
Wouldn't `assert_script_run` be better here ? `script_run` gives only RC but does not fail with command failure.

Reply:
I don't think `partprobe` return non-zero retvals, so it will be the same. Changing it to `enter_cmd` could be an idea, but I think I prefer this with `script_run` or `assert_script_run` to force the API to call on `wait_serial` and wait for a little bit.

Reply:
I don't think `partprobe` returns non-zero retvals, so it will be the same. Changing it to `enter_cmd` could be an idea, but I think I prefer this with `script_run` or `assert_script_run` to force the API to call on `wait_serial` and wait for a little bit more.

Reply:
Ah I see, ok.